### PR TITLE
Fix including pywintypes*.dll and pywincom*.dll

### PR DESCRIPTION
**Pull request by compholio**
This is the issue from the comment:
https://github.com/albertosottile/py2exe/issues/27#issuecomment-553817486

Hopefully the way I've approached this is alright.

### CODE REVIEW

File: py2exe/mf3.py [hunk 1]
```diff
@@ -684,6 +691,27 @@ def __dest_file__(self):
...
+        return self.__file__
+
+
+    @property
+    def __extension__(self):
+        if self.__loader__.__class__ != importlib.machinery.ExtensionFileLoader:
+            return None
+        if self.__file__.endswith(EXTENSION_TARGET_SUFFIX):
+            return EXTENSION_TARGET_SUFFIX
+        if 'pywintypes' in self.__file__:
```
Comment (albertosottile) [inline py2exe/mf3.py]: I am not super convinced by this, it feels like a hack. I did not have the time to investigate why these DLLs are picked in the module and not as external libraries, which is causing this issue in the first place.

Comment (compholio) [reply py2exe/mf3.py]: Ah, the issue that is causing it in the first place is that `determine_dll_type` sees a linkage to the python DLL in the dependencies.  I checked with Dependency Walker and this linkage is real.

Another choice is to check for pywintypes and pywincom in `determine_dll_type` and treat them as a regular DLL, but my concern with that approach is that py2exe would not build a wrapper such that "import" would work properly on those modules:
https://github.com/albertosottile/py2exe/blob/master/py2exe/runtime.py#L443

**Comment by alex-the-man** [comment]
I'm hitting this issue as well. Should we change EXTENSION_TARGET_SUFFIX to include both .pyd and .dll?

**Comment by albertosottile** [comment]
`pywin32` is included in the CI/CD test routine and it is working well in the 0.10.0.x tree without this fix. It might be that the applied changes in `dllfinder` solved also this issue.

Can you try to reproduce this bug with the latest version (0.10.0.2)? If you manage, I would like to have a script that reproduces  that I can integrate in my test chain. 

Thank you all for your time.

**Comment by albertosottile** [comment]
Closing this PR as `pywin32` functionality is verified in CI and no further information has been submitted lately. Please feel free to reopen it if you experience the same problem again. Nevertheless, I would like to encourage you to not use `conda` when bundling with `py2exe`, as that environment is not supported. 

If you wish to have a clean and reproducible build environment, I would advise you to use Docker.

**Comment by albertosottile** [comment]
The upcoming version 0.10.2.0 will introduce a new module finder that, in principle, should not trigger your issue anymore. Preliminary wheels for this version can be found on Bintray here: https://bintray.com/py2exe/py2exe/py2exe/0.10.2.0#files

@compholio @alex-the-man  It would be great if you could install this version, repeat your tests with it, and report here if you still experience this problem. Thanks for your time.

**Comment by compholio** [comment]
@albertosottile I'm not working where I was before, so I cannot check with a complicated project, but it at least builds something simple.  So, I'd say this is likely fixed.

**Pull request closed without merging**